content/bd/0001-intro-dbms.md

```diff
@@ -0,0 +1,253 @@
+  na base de dados e como se relacionam entre si. Em SGBDs relacionais, isto é
+  aquilo a que casualmente nos referimos como _schema_.
+
+- **Nível Físico/Interno**: É responsável pelo armazenamento dos dados, isto é,
+  a sua representação física no sistema.
+
+:::info[Exemplo]
+
+Imaginemos que temos o esquema lógico abaixo:
+
+```
+Student (ist_id: string, student_name: string, birthday: date)
+Course (course_id: string, course_name: string,
+        description: integer)
+Shift (shift_id: string, course_id: string, capacity: integer)
+Enrollment (ist_id: string, shift_id: string)
+Room (room_name: string, course_id: string, capacity: integer)
+```
+
+Uma possível _view_ para este esquema lógico seria:
+
+```
+CourseShiftRooms (room_name: string, course_name: string,
+                  enrolled_count: integer)
+```
+
+Podemos reparar que esta relação não está representada explicitamente no
+esquema lógico, no entanto todos os dados podem ser derivados das relações
+definidas no mesmo.
+
+:::
+
+A [**grande vantagem**](color:green) desde modelo é a independência dos dados. Como estamos
+a dividir a estrutura em três níveis, podemos alterar cada um deles sem ter de propagar
+estas alterações "para cima":
+
+- Podemos alterar o esquema lógico sem alterar as nossas aplicações, visto que as vistas
+  se podem manter.
+- Podemos alterar a organização física dos dados, isto é, como são armazenados, sem termos
+  de modificar o esquema lógico.
+
+## Modelos de Dados
+
+Um SGBD tem de usar um **modelo de dados**, um conjunto de conceitos e métodos, para descrever
+**dados**, **relacionamentos**, **semântica de dados** e **restrições**.
+
+Existem vários tipos de modelos de dados hoje em dia:
+
+- **Relacional** (o mais usado)
+- **Entidade-Relação** (para conceção de bases de dados)
+- **Orientada a Objetos**
+- **Semi-Estruturados** (XML, JSON, etc.)
+
+### Modelo Relacional
+
+Atualmente, o **modelo relacional** é o mais usado.
+Como o próprio nome indica, é composto por relações.
+Uma relação (chamada no dia-a-dia de tabela/_table_), é constituída pelo seu **nome**,
+o **nome de cada atributo** (_field_) e o **tipo de cada atributo**.
+
+:::info[Exemplo]
+
+```
+Student (ist_id: string, student_name: string, birthday: date)
+```
+
+Podemos depois adicionar linhas (_rows_) nesta relação. Neste caso, cada linha
+representa um único aluno.
+
+| `ist_id`   | `student_name` | `birthday` |
+| ---------- | -------------- | ---------- |
+| ist1123456 | Diogo          | 2002-03-06 |
+| ist1123234 | Rafa           | 2002-08-05 |
+| ist1124453 | Tomás          | 2002-12-18 |
+| ist1123534 | Diogo          | 2001-09-16 |
+| ist1123532 | Diogo          | 2002-04-27 |
+
+:::
+
+Neste exemplo podemos querer também enforçar restrições de integridade.
+Neste caso, a restrição mais óbvia é garantir que não existem dois alunos diferentes
+com o mesmo `ist_id`. No entanto, podem existir alunos com o mesmo nome.
```

Comment:
I don't get the purpose of these lines; they're motivation for something that's never concretized into actual content. It's too early for this specifically (SQL hasn't even been introduced yet), but it feels like a sentence of the sort of
```
Assim, devemos usar, por exemplo a _constraint_ `UNIQUE` na definição da coluna `ist_id` mas não na da coluna `student_name`.
```
should follow to conclude the premises already present.

Reply:
I'll just say that we'll look at this later, since I don't want to introduce SQL unique/primary key constraints here as you said